pkg/controller/controller.go

```diff
@@ -1363,24 +1388,53 @@ func (s *Service) ControllerExpandVolume(ctx context.Context, req *csi.Controlle
 		return nil, status.Errorf(codes.OutOfRange, "volume exceeds allowed limit")
 	}
 
+	array, ok := s.Arrays()[arrayID]
+	if !ok {
+		return nil, status.Errorf(codes.InvalidArgument, "unable to find array with ID %s", arrayID)
+	}
+	client := array.Client
+
 	if protocol == "scsi" {
-		vol, err := s.Arrays()[arrayID].Client.GetVolume(ctx, id)
+		vol, err := client.GetVolume(ctx, id)
 		if err != nil {
-			return nil, status.Errorf(codes.OutOfRange, "detected SCSI protocol but wasn't able to fetch the volume info")
+			return nil, status.Error(codes.NotFound, "detected SCSI protocol but wasn't able to fetch the volume info")
 		}
+
 		if vol.Size < requiredBytes {
-			_, err = s.Arrays()[arrayID].Client.ModifyVolume(context.Background(), &gopowerstore.VolumeModify{Size: requiredBytes}, id)
+			// get replication session ID if Metro is configured. Pause the session and resume after updating size.
+			replicationSessionID, err := GetMetroSessionID(ctx, client, id, vol, remoteArrayID)
 			if err != nil {
-				return nil, err
+				return nil, status.Error(codes.Internal, "error while identifying metro configuration on volume")
+			}
+
+			if replicationSessionID != "" {
```

Comment:
with the change to `vg.MetroReplicationSessionID` we can just check for an empty string this at the top of the `if` block and return an error if empty. We should also be able to get rid of any additional checks further down